plugin/pulpcore/plugin/stages/declarative_version.py

```diff
@@ -81,33 +81,36 @@ def __init__(self, first_stage, repository, mirror=True):
                  :class:`~pulpcore.plugin.stages.DeclarativeVersion stream`, and does not remove any
                  pre-existing units in the :class:`~pulpcore.plugin.models.RepositoryVersion`.
                  'True' is the default.
+             download_artifacts (bool): 'True' includes Artifact downloading and saving along with
+                 content unit query and saving. 'False' will only handle content units and will not
+                 perform Artifact downloading and saving as part of the pipeline. 'False' is the
+                 default.
         """
         self.first_stage = first_stage
         self.repository = repository
         self.mirror = mirror
+        self.download_artifacts = download_artifacts
 
-    def pipeline_stages(self, new_version):
```

Comment:
Actually, `new_version` is here by intention if a plugin writer wants to insert a stage that uses it.

And I might even have a use case for it now: As said in my email on pulp-dev, I would like to avoid that content from other repositories show up during a sync of a repo (unless it is really the same content of course). If the upstream does not provide digests, one way to achieve this is to look at the current content in the base version when querying for existing content (like the sync in pulp_ansible does using ChangeSet https://github.com/pulp/pulp_ansible/blob/83f377602897ebf58261ca0d92473dc57a64ac25/pulp_ansible/app/tasks/synchronizing.py#L72-L75).

Reply:
This makes good sense to me. I'm bringing this back now. Thanks!